pkg/reconciler/instances/istio/bootstrap.go

```diff
@@ -75,44 +75,64 @@ func newDefaultCommanderResolver(paths []string, log *zap.SugaredLogger) (action
 	}, nil
 }
 
-// parsePaths func parses and validates executable paths. The input must contain a list of full/absolute filesystem paths of binaries, separated by a colon character ':'
+// parsePaths func parses and validates executable paths. The input must contain a list of full/absolute filesystem paths of binaries, separated by a semicolon character ';'
 // isValid function is used to validate every single binary path in the input.
-func parsePaths(input string, isValid func(string) error) ([]string, error) {
+func parsePaths(input string, isValid func(string, *zap.SugaredLogger) error, logger *zap.SugaredLogger) ([]string, error) {
 	trimmed := strings.TrimSpace(input)
 	if trimmed == "" {
 		return nil, errors.Errorf("%s env variable is undefined or empty", istioctlBinaryPathEnvKey)
 	}
 	if len(trimmed) > istioctlBinaryPathMaxLen {
 		return nil, errors.New(fmt.Sprintf("%s env variable exceeds the maximum istio path limit of %d characters", istioctlBinaryPathEnvKey, istioctlBinaryPathMaxLen))
 	}
-	pathDefs := strings.Split(trimmed, ":")
+	pathDefs := strings.Split(trimmed, ";")
 	var res []string
 	for _, path := range pathDefs {
 		val := strings.TrimSpace(path)
 		if val == "" {
 			return nil, errors.New("Invalid (empty) path provided")
 		}
-		if err := isValid(val); err != nil {
+		if err := isValid(val, logger); err != nil {
 			return nil, err
 		}
 		res = append(res, val)
 	}
 	return res, nil
 }
 
-// TODO: test it
-func validatePath(path string) error {
+func validatePath(path string, logger *zap.SugaredLogger) error {
```

Comment:
The contract for this function before was to return an error if file was corrupted or not executable. If you want to change it then please adjust the name appropriately as well so it reflects its real purpose. I would suggest: `ensureExecutable` (we don't need to repeat `path` imho as it is already a function parameter) or `ensureFileExecutable` or even `ensureFileExecutableFrom` (implicitly from path when you keep reading function parameters too)